### DISCUSSION TITLE
How to extract XML documentation comments?

### DISCUSSION THREAD

**Discussion by gpetrou** [dotnet/runtime#48264] [Q&A] [answered]
With .NET Framework, we can check "Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0" path for the XML comments of each DLL. This can be used for example with IronPython to display the comments for a method as implemented in https://github.com/IronLanguages/ironpython3/blob/master/Src/IronPython/Runtime/Types/DocBuilder.cs
If we use the following code:

    import clr 
    clr.AddReference("System") 
    import System 
    import System.Text 
    print (System.Text.StringBuilder.__doc__)

We will get:

    Represents a mutable string of characters. This class cannot be inherited.

    StringBuilder()
    StringBuilder(capacity: int)
    StringBuilder(value: str)
    StringBuilder(value: str, capacity: int)
    StringBuilder(value: str, startIndex: int, length: int, capacity: int)
    StringBuilder(capacity: int, maxCapacity: int)

With .NET Core, looking at C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.12 there are no XML files. How can we extract the XML comments in this case? Are we supposed to download them from somewhere?

**Comment by billrob** [comment, marked as answer]
@gpetrou I found mine at:  C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\3.1.0\ref\netcoreapp3.1

From information I found here: https://docs.microsoft.com/en-us/dotnet/core/install/localized-intellisense

**Reply by gpetrou** [reply]
Thank you for the reply.
Looking at the ref folders I can see many XML documentation files.
Unfortunately, I still don't see System.Private.CoreLib.xml that should have the StringBuilder information.
I have created https://github.com/dotnet/installer/issues/9729 to see if this could be fixed.

**Reply by billrob** [reply]
Did you try searching for it?

I found it in System.Runtime.xml

```
PS C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\3.1.0\ref> findstr /s /i /p /c:"Represents a mutable string of characters. This class cannot be inherited." *.*
netcoreapp3.1\System.Runtime.xml:      <summary>Represents a mutable string of characters. This class cannot be inherited.</summary>
```

**Reply by gpetrou** [reply]
The way I search for the documentation file is by checking the StringBuilder Type Assembly property which returns System.Private.CoreLib.dll. So, following the convention from .NET Framework, I should be able to find the documentation in System.Private.CoreLib.xml. I am not only after StringBuilder, but any type defined in the libraries.
